File: AlbumLibrary/src/main/java/com/album/ui/widget/SimpleGridDivider.java
package com.album.ui.widget;

import android.graphics.Rect;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;

/**
 * by y on 21/08/2017.
 */

public class SimpleGridDivider extends RecyclerView.ItemDecoration {

    private final int divider;

    public SimpleGridDivider(int divider) {
        this.divider = divider;
    }

    @Override
    public void getItemOffsets(Rect outRect, View view, RecyclerView parent, RecyclerView.State state) {
        int position = parent.getChildAdapterPosition(view);
        GridLayoutManager layoutManager = (GridLayoutManager) parent.getLayoutManager();
        int spanCount = layoutManager.getSpanCount();
        int i = position % spanCount;
        boolean first = (position / spanCount + 1) == 1;
        int top = first ? divider / 2 : 0;
        int left = i + 1 == 1 ? divider : divider / 2;
        int right = i + 1 == spanCount ? divider : divider / 2;
        outRect.set(left, top, right, divider);

    }
}

File: AlbumLibrary/src/main/java/com/album/widget/SimpleGridDivider.kt
package com.album.widget

import android.graphics.Rect
import android.view.View
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView

/**
 * @author y
 * @create 2019/2/27
 */
internal class SimpleGridDivider(private val divider: Int) : RecyclerView.ItemDecoration() {
    override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State) {
        super.getItemOffsets(outRect, view, parent, state)
        val position = parent.getChildAdapterPosition(view)
        val layoutManager = parent.layoutManager as GridLayoutManager
        val spanCount = layoutManager.spanCount
        val i = position % spanCount
        val first = position / spanCount + 1 == 1
        val top = if (first) divider / 2 else 0
        val left = if (i + 1 == 1) divider else divider / 2
        val right = if (i + 1 == spanCount) divider else divider / 2
        outRect.set(left, top, right, divider)
    }
}